grapple/middleware.py

```diff
@@ -48,6 +36,6 @@ def resolve(self, next, root, info: ResolveInfo, **args):
         parent_name = info.parent_type.name
         if field_name in self.field_middlewares and parent_name in ROOT_TYPES:
             for middleware in self.field_middlewares[field_name]:
-                return middleware(next, root, info, **args)
+                return middleware(self, next, root, info, **args)
```

Comment:
would be good to add a note as to why we're passing `self` here, perhaps with a link to docs or code

Reply:
I stuck that in because I was getting a parameter mismatch. I'm not exactly sure why the change was required and didn't find any docs for the change.  I'll spend some time trying to figure it out, but I'm not really familiar with how the middleware works.

Reply:
I did a little more digging and https://github.com/graphql-python/graphene/issues/812 is the most related discussion I can find about why self needs to be there... but what I really don't know is what GrappleMiddleware is intended to do. A deeper change may be necessary for it to work propertly.  I'm pretty sure I'm passing the wrong self into resolve there, which could cause issues with some middleware that expect to have something on their self.